src/pages/CDP.js

```diff
@@ -114,21 +117,59 @@ const CdpViewCard = ({ title, rows, isAction }) => {
   );
 };
 
-function CDPView({ cdpId }) {
+function CDPView({ cdpTypeSlug, getIlk }) {
   const { maker } = useMaker();
 
   // TODO cdpTypeSlug should become `id` or we should have both cdpTypeSlug AND id.
   const [cdpState, setCDPState] = useState(null);
+  const [collateralVal, setCollateralVal] = useState('');
+  const [debtVal, setDebtVal] = useState('');
+  const [ilk, setIlk] = useState(null);
 
   useEffect(() => {
     (async () => {
       const cdpManager = maker.service('mcd:cdpManager');
-      const cdp = await cdpManager.getCdp(cdpId);
+      const cdp = await cdpManager.getCdp(parseInt(cdpTypeSlug));
       setCDPState(cdp);
+      setIlk(getIlk(cdp.ilk));
+      setCollateralVal(await cdp.getCollateralValue());
+      setDebtVal(await cdp.getDebtValue());
```

Comment:
We might be better off doing only one `setState` given we don't benefit from any re-renders caused by updated state before all is available.

```js
   const cdp = await cdpManager.getCdp(parseInt(cdpTypeSlug));
   const ilk = getIlk(cdp.ilk)
   const debt = await cdp.getDebtValue();
   const collateral = await cdp.getCollateralValue();
   setCDP({
    ...cdp,
    ilk,
    debt,
   collateral
  })
```